addons/web/controllers/report.py

```diff
@@ -84,7 +84,10 @@ def report_barcode(self, barcode_type, value, **kwargs):
         except (ValueError, AttributeError):
             raise werkzeug.exceptions.HTTPException(description='Cannot convert into barcode.')
 
-        return request.make_response(barcode, headers=[('Content-Type', 'image/png')])
+        return request.make_response(barcode, headers=[
+            ('Content-Type', 'image/png'),
+            ('Cache-Control', f'public, max-age={http.STATIC_CACHE_LONG}, immutable'),
```

Comment:
OOC how does it work actually ? Can't we leak some information with this ?

Reply:
Take the following example, I uploaded a private image on a runbot at URL:

https://56860141-master-all.runbot138.odoo.com/web/image/2913?width=64&height=64
![](https://56860141-master-all.runbot138.odoo.com/web/image/2913?&width=64&height=64)

You can see it only with an access-token:
https://56860141-master-all.runbot138.odoo.com/web/image/2913?access_token=c119889a-d06b-4971-ac8e-95b89892c66d&width=64&height=64
![](https://56860141-master-all.runbot138.odoo.com/web/image/2913?access_token=c119889a-d06b-4971-ac8e-95b89892c66d&width=64&height=64)

If I was to use that precise URL and generate a qr-code, it would fail because the `?width=...&height=...&access_token=...` are wrongly captured by the `/barcode/QR` route and removed by the controller:

https://www.odoo.com/report/barcode/QR/https://56860141-master-all.runbot138.odoo.com/web/image/2913?access_token=c119889a-d06b-4971-ac8e-95b89892c66d&width=64&height=64
![](https://www.odoo.com/report/barcode/QR/https://56860141-master-all.runbot138.odoo.com/web/image/2913?access_token=c119889a-d06b-4971-ac8e-95b89892c66d&width=64&height=64)
![image](https://github.com/odoo/odoo/assets/8208953/baa2c222-9cdc-435b-a3d8-335d5c173729)

But using advanced URL knowledge we can leverage this limitation by encoding `?`, `&` and `=`:
https://www.odoo.com/report/barcode/QR/https%3A//56860141-master-all.runbot138.odoo.com/web/image/2913%3Faccess_token%3Dc119889a-d06b-4971-ac8e-95b89892c66d%26width%3D64%26height%3D64?width=256&height=256
![](https://www.odoo.com/report/barcode/QR/https%3A//56860141-master-all.runbot138.odoo.com/web/image/2913%3Faccess_token%3Dc119889a-d06b-4971-ac8e-95b89892c66d%26width%3D64%26height%3D64?width=256&height=256)
![image](https://github.com/odoo/odoo/assets/8208953/d67f7b9d-67ac-4545-8df5-d021c1c0620b)

Thus access the protected image via the qrcode and the access token. Accessing the final image is still subject for authorization by odoo.com, the *pirate* image itself isn't put in public cache.

Reply:
Ok, thanks for detailed explanation :) it generates the code for the link, hence don't use it for private links (or at your own risks).